### ISSUE TITLE
Downloaded gzip is not in proper format

### ISSUE THREAD

**Issue by snxfz947** [snxfz947/unix-server-onedrive-client#14]
```
tar xvzf 1557212401_2019-05-07-00-00-01_private_repos.tar.gz

gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now
```

**Comment by snxfz947** [comment]
Seems to still work with directories < 60 MB and in between 60 MB and 120 MB - the specific directory that threw this error was ~260 MB - not sure if it's the size or the content that causes the gzip issues

**Comment by snxfz947** [comment]
Successful:
```
 Thu May  9 00:00:59 2019 [VERBOSE]  UPLOAD:     Uploading {'Content-Range': 'bytes 0-28442479/154271106', 'Content-  Length': '28442480'}
 Thu May  9 00:01:35 2019 [VERBOSE]  UPLOAD:     Uploading {'Content-Range': 'bytes 28442480-91356865/154271106',
 'Content-Length': '62914386'}
 Thu May  9 00:02:51 2019 [VERBOSE]  UPLOAD:     Uploading {'Content-Range': 'bytes 91356866-154271105/154271106',
 'Content-Length': '62914240'}
```

Unsuccessful:
```
 Thu May  9 00:17:57 2019 [VERBOSE]  UPLOAD:     Uploading {'Content-Length': '62914386', 'Content-Range': 'bytes 0-  62914385/154271106'}
 Thu May  9 00:20:15 2019 [VERBOSE]  UPLOAD:     Uploading {'Content-Length': '28442480', 'Content-Range': 'bytes
 125828626-154271105/154271106'}
 Thu May  9 00:20:21 2019 [VERBOSE]  UPLOAD:     Uploading {'Content-Length': '28442480', 'Content-Range': 'bytes
 62914386-91356865/154271106'}
 Thu May  9 00:21:28 2019 [VERBOSE]  UPLOAD:     Uploading {'Content-Length': '62914240', 'Content-Range': 'bytes
 91356866-154271105/154271106'}
```

**Comment by snxfz947** [comment]
Seems to be the ordering - `get_chunks()` is returning an arbitrary order - this should be sorted

**Comment by snxfz947** [comment]
Fixed with 74bb0af942a6f6a71d237b82c840f7f9e07b852b
